### ISSUE TITLE
PandasParallelLFApplier does not run in parallel when using Spacy as preprocessor

### ISSUE THREAD

**Issue by hardianlawi** [snorkel-team/snorkel#1662]
## Issue description

`PandasParallelLFApplier` does not improve the speed when using `spacy` as preprocessing step.

## Code example/repro steps

No parallelization

```python
import spacy
import pandas as pd
from timeit import default_timer
from snorkel.labeling import PandasLFApplier, labeling_function
from snorkel.preprocess import preprocessor


nlp = spacy.load("en_core_web_sm")


@preprocessor(memoize=True)
def spacy_doc(x):
    doc = nlp(x.text)
    return doc


@labeling_function(pre=[spacy_doc])
def greater_than_10(x):
    for token in x:
        pass
    if int(token.text) > 10:
        return 1
    return 0


df = pd.DataFrame({"text": [f"This is sentence {num}" for num in range(2000)]})
applier = PandasLFApplier([greater_than_10])

start = default_timer()
L = applier.apply(df)
end = default_timer()

print(end - start) # 10.994037621989264 seconds
```

With parallelization

```python
import spacy
import pandas as pd
from timeit import default_timer
from snorkel.labeling import labeling_function, PandasLFApplier
from snorkel.labeling.apply.dask import PandasParallelLFApplier
from snorkel.preprocess import preprocessor


nlp = spacy.load("en_core_web_sm")


@preprocessor(memoize=True)
def spacy_doc(x):
    doc = nlp(x.text)
    return doc


@labeling_function(pre=[spacy_doc])
def greater_than_10(x):
    try:
        if int(x[-1].text) > 10:
            return 1
    except Exception as e:
        return 0
    return 0


df = pd.DataFrame({"text": [f"This is sentence {num}" for num in range(2000)]})
applier = PandasParallelLFApplier([greater_than_10])

start = default_timer()
L = applier.apply(df, n_parallel=4)
end = default_timer()

print(end - start) # 17.868044445000123
```

## Expected behavior

I would expect `PandasParallelLFApplier` to improve the speed at least a little bit even if it's not much

## System info

* How you installed Snorkel (conda, pip, source): pip
* OS: Ubuntu 20.04
* Python version: 3.8
* Snorkel version: 0.9.7
* Versions of any other relevant libraries:

## Additional context

I think the problem is because the spacy library is not sharable between processes(?). I would love to know the workaround for this if anyone has a solution (with or without parallelization).

**Comment by hardianlawi** [comment]
Ultimately, the thing that I care about the most is to improve the speed. I'm wondering if it makes sense to have something like `BatchPandasLFApplier`. `spacy` model provides`nlp.pipe` method which could boost the speed by passing batch of inputs, but currently snorkel isn't really using it.

**Comment by hardianlawi** [comment]
@bhancock8 Sure. I'll send a PR for this. In the meantime, I have created another PR to bump up the tensorboard version and I need someone to review it for me as I'm not sure why the CI is failing 🙏🏼 .
